Kudu.Core/Jobs/JobsManagerBase.cs

```diff
@@ -495,6 +496,16 @@ private static string FindCommandToRun(FileInfoBase[] files, out IScriptHost scr
                         }
                     }
                 }
+
+                foreach (var supportedFileName in scriptHost.SupportedFileNames)
```

Comment:
There is a functional test [`JobsScriptsShouldBeChosenByOrder()`](https://github.com/projectkudu/kudu/blob/master/Kudu.FunctionalTests/Jobs/WebJobsTests.cs#L697-L747) for this logic. I'm adding coverage for the `DnxScriptHost`